pkgs/development/libraries/qt-6/default.nix

```diff
@@ -163,26 +169,32 @@ let
         inherit (darwin.apple_sdk_11_0.frameworks) WebKit;
       };
 
-      wrapQtAppsHook = makeSetupHook
+      wrapQtAppsHook = callPackage ({ makeBinaryWrapper }: makeSetupHook
         {
           name = "wrap-qt6-apps-hook";
-          propagatedBuildInputs = [ buildPackages.makeBinaryWrapper ];
-        } ./hooks/wrap-qt-apps-hook.sh;
+          propagatedBuildInputs = [ makeBinaryWrapper ];
+        } ./hooks/wrap-qt-apps-hook.sh) { };
 
-      qmake = makeSetupHook
+      qmake = callPackage ({ qtbase }: makeSetupHook
         {
           name = "qmake6-hook";
-          propagatedBuildInputs = [ self.qtbase.dev ];
+          propagatedBuildInputs = [ qtbase.dev ];
           substitutions = {
             inherit debug;
             fix_qmake_libtool = ./hooks/fix-qmake-libtool.sh;
           };
-        } ./hooks/qmake-hook.sh;
+        } ./hooks/qmake-hook.sh) { };
+    } // lib.optionalAttrs config.allowAliases {
+      # Convert to a throw on 03-01-2023 and backport the change.
```

Comment:
Not sure when is exactly 03-01-2023, but isn't it already passed?

Reply:
Meant 2024 :P

I have it in my calendar so no need to fix the typo